handlers/abtest/abtesthandler.go

```diff
@@ -0,0 +1,62 @@
+package abtest
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/ONSdigital/dp-cookies/cookies"
+)
+
+const (
+	SearchTestCookieAspect = "dsa-search"
+	SearchNewExit          = "exit-new-search"
+
+	RelcalTestCookieAspect = "dsa-release-calendar"
+	RelcalNewExit          = "exit-new-relcal"
+)
+
+func Handler(abTest bool, new, old http.Handler, percentage int, aspectID, domain, exitNew string) http.Handler {
+	if abTest {
+		return ABTestHandler(new, old, percentage, aspectID, domain, exitNew)
+	} else {
+		return ABTestPurgeHandler(new, aspectID, domain)
+	}
+}
+
+// ABTestHandler routes requests to either the old or new handler, for a given aspectID, according to the given percentage
+// i.e. for the given percentage of calls X, X% will be routed to the new handler, and the remainder to the old handler.
+// Most of the functionality is provided by the dp-cookies library, which uses a single ab_test cookie to embed all aspects
+// If the aspect does not exist or has expired, it is created/renewed according to a particular randomiser - in general
+// the DefaultABTestRandomiser in the library is sufficient
+// A well known string - the exitNew string -  can be used as a query parameter to the call, in order to definitively chose
+// the old handler
+func ABTestHandler(new, old http.Handler, percentage int, aspectID, domain, exitNew string) http.Handler {
```

Comment:
I think the visibility of these 2 functions could now be reduced and have only `Handler` as public.